### PR TITLE
AI lawset can't be asimov on unique-lawset station trait shifts

### LINKED ISSUE #61705: The Unique AI station trait can, and often does, give Asimov
## Round ID: [170251](https://scrubby.melonmesa.com/round/170251)

The Unique AI code picks from law_weight, which Asimov is king in. That means most of the time, you will get Asimov again, which is really lame.

### PR DESCRIPTION
**Pull request by Mooshimi**
## About The Pull Request
-adds a test to the loop that picks a random lawset for the ai on unique lawset rounds that doesnt let it get asimov again
Fixes: #61705 

## Why It's Good For The Game
-asimov boring, event with different laws fun

## Changelog

:cl:
code: AIs will no longer get the asimov lawset on unique-lawset station trait rounds
/:cl:

### CODE REVIEW

File: code/datums/ai_laws.dm [hunk 1]
```diff
@@ -263,6 +263,7 @@
 /datum/ai_laws/proc/pick_weighted_lawset()
 	var/datum/ai_laws/lawtype
 	var/list/law_weights = CONFIG_GET(keyed_list/law_weight)
+	law_weights -= /datum/ai_laws/default/asimov
```
Comment (Ghommie) [inline code/datums/ai_laws.dm:266]: First, this won't fix it. This is a list of lawsets ids and not types. so it ought to be "asimov", not  `/datum/ai_laws/default/asimov`.
Second, it should be a blacklist arg containing ai_laws typepaths. Their initial id value will be removed from the law_weights this way:
```
	for(var/lawset in blacklist)
		law_weights -= initial(lawset.id)
```
Third, the arg should be unset by default and set to list("asimov") in this call:
https://github.com/tgstation/tgstation/blob/55b92786dcf0f9c5172efd111a6ecbf83ec0dae2/code/datums/ai_laws.dm#L234

**Review by Ghommie — CHANGES_REQUESTED** [review]

File: code/datums/ai_laws.dm (hunk 1 above)
Comment (Mooshimi) [reply code/datums/ai_laws.dm]: > Third, the arg should be unset by default and set to `list(/datum/ai_laws/default/asimov)` in this call (the proc is called three times in the code and one of them is a config setting for random weighted lawsets):
> 

is what I did what you meant by having it unset? I wasn't certain on your meaning by that.

File: code/datums/ai_laws.dm [hunk 2]
```diff
@@ -3,7 +3,7 @@
 #define LAW_SUPPLIED "supplied"
 #define LAW_ION "ion"
 #define LAW_HACKED "hacked"
-
+#define AI_LAWS_ASIMOV "asimov"
```
Comment (Mothblocks) [inline code/datums/ai_laws.dm:6]: `#undef` this at the very end of the file, like this:

```dm
#undef AI_LAWS_ASIMOV
```

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: code/datums/ai_laws.dm (hunk 1 above)
Comment (Ghommie) [reply code/datums/ai_laws.dm]: Not exactly. An argument is in layman terms what you see between the parenthesis of a proc definition (e.g. `thing/proc/does_something(argument)`) or call (`thing.does_something(arg_value)`). Though it doesn't matter anymore since the way it's done now works too.
